src/script/media/MediaDevicesHandler.ts

```diff
@@ -165,15 +199,33 @@ export class MediaDevicesHandler {
    */
   private subscribeToObservables(): void {
     this.currentDeviceId.audioInput.subscribe(mediaDeviceId => {
+      const faveAudioInput = loadValue<string[]>(FavoriteDeviceTypes.AUDIO_INPUT) ?? [];
+      const newFaveAudioInput =
```

Comment:
The structure for faveAudioInput, faveAudioOutput, and faveVideoInput is repetitive. We can create a reusable function to achieve the same result.

```
const getNewFavorites = (deviceType, mediaDeviceId) => {
  const favorites = loadValue<string[]>(deviceType) ?? [];
  const previousDeviceSupportCount = this.previousDeviceSupport[deviceType];
  const availableDevicesCount = this.availableDevices[deviceType]().length;

  if (previousDeviceSupportCount >= availableDevicesCount) {
    return [...favorites.filter(id => id !== mediaDeviceId), mediaDeviceId];
  }
  return [mediaDeviceId, ...favorites.filter(id => id !== mediaDeviceId)];
};

```

Reply:
const newFaveAudioInput = getNewFavorites(FavoriteDeviceTypes.AUDIO_INPUT, mediaDeviceId);
const newFaveAudioOutput = getNewFavorites(FavoriteDeviceTypes.AUDIO_OUTPUT, mediaDeviceId);
const newFaveVideoInput = getNewFavorites(FavoriteDeviceTypes.VIDEO_INPUT, mediaDeviceId);